cosmwasm/packages/wasmi-runtime/src/cosmwasm/types.rs

```diff
@@ -104,27 +105,130 @@ pub struct Coin {
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
-#[serde(rename_all = "lowercase")]
-pub enum CosmosMsg {
+#[serde(untagged)]
+pub enum WasmOutput {
+    ErrObject {
+        #[serde(rename = "Err")]
+        err: Value,
+    },
+    OkString {
+        #[serde(rename = "Ok")]
+        ok: String,
+    },
+    OkObject {
+        #[serde(rename = "Ok")]
+        ok: OkObject,
+    },
+}
```

Comment:
You can rewrite this as:
```rust
#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
pub enum WasmOutput {
  Ok(OkObjectOrString),
  Err(Value)
}

#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
#[serde(untagged)]
pub enum OkObjectOrString {
  Str(String),
  Obj(OkObject),
}
```
Which should generate the exact same parser.
Then again the way you did it makes the match later nicer...
So go with what you think is better - after writing this i like your way better, just wanted to show you you could simplify the definition here.

Reply:
Yep I think the current way is more readable